homeassistant/components/sensor/syncthru.py

```diff
@@ -0,0 +1,236 @@
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+
+REQUIREMENTS = ['pysyncthru==0.3.0.1']
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_NAME = 'Samsung Printer'
+DEFAULT_MONITORED_CONDITIONS = [
+    'toner_black',
+    'toner_cyan',
+    'toner_magenta',
+    'toner_yellow',
+    'drum_black',
+    'drum_cyan',
+    'drum_magenta',
+    'drum_yellow',
+    'tray_1',
+    'tray_2',
+    'tray_3',
+    'tray_4',
+    'tray_5',
+    'output_tray_0',
+    'output_tray_1',
+    'output_tray_2',
+    'output_tray_3',
+    'output_tray_4',
+    'output_tray_5',
+]
+COLORS = [
+    'black',
+    'cyan',
+    'magenta',
+    'yellow'
+]
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_RESOURCE): cv.url,
+    vol.Optional(
+        CONF_NAME,
+        default=DEFAULT_NAME
+    ): cv.string,
+    vol.Optional(
+        CONF_MONITORED_CONDITIONS,
+        default=DEFAULT_MONITORED_CONDITIONS
+    ): vol.All(cv.ensure_list, [vol.In(DEFAULT_MONITORED_CONDITIONS)])
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the SyncThru component."""
+    from pysyncthru import SyncThru, test_syncthru
+
+    if discovery_info is not None:
+        host = discovery_info.get(CONF_HOST)
+        name = discovery_info.get(CONF_NAME, DEFAULT_NAME)
+        _LOGGER.debug("Discovered a new Samsung Printer: %s", discovery_info)
+        # Test if the discovered device actually is a syncthru printer
+        if not test_syncthru(host):
+            _LOGGER.error("No SyncThru Printer found at %s", host)
+            return
+        monitored = DEFAULT_MONITORED_CONDITIONS
+    else:
+        host = config.get(CONF_RESOURCE)
+        name = config.get(CONF_NAME)
+        monitored = config.get(CONF_MONITORED_CONDITIONS)
+
+    # Main device, always added
+    try:
+        printer = SyncThru(host)
+    except BaseException:
+        # if an exception is thrown, printer cannot be set up
+        return
+    printer.update()
+    devices = [SyncThruMain(hass, printer, name)]
+
+    for key in printer.tonerStatus(filter_supported=True):
+        if 'toner_{}'.format(key) in monitored:
+            devices.append(SyncThruToner(hass, printer, name, key))
+    for key in printer.drumStatus(filter_supported=True):
+        if 'drum_{}'.format(key) in monitored:
+            devices.append(SyncThruDrum(hass, printer, name, key))
+    for key in printer.inputTrayStatus(filter_supported=True):
+        if 'tray_{}'.format(key) in monitored:
+            devices.append(SyncThruInputTray(hass, printer, name, key))
+    for key in printer.outputTrayStatus():
+        if 'output_tray_{}'.format(key) in monitored:
+            devices.append(SyncThruOutputTray(hass, printer, name, key))
+
+    add_devices(devices, True)
+    return
+
+
+class SyncThruSensor(Entity):
+    """Implementation of an abstract Samsung Printer sensor platform."""
+
+    def __init__(self, hass, syncthru, name):
```

Comment:
Don't pass in `hass`. It will be set on the entity when it has been added to home assistant.